hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/shell/BulkDeleteCommand.java

```diff
@@ -134,10 +138,14 @@ void deleteInBatches(BulkDelete bulkDelete, List<Path> paths) throws IOException
     while (batches.hasNext()) {
       try {
         List<Map.Entry<Path, String>> result = bulkDelete.bulkDelete(batches.next());
-        LOG.debug("Deleted Result:{}", result.toString());
+        if(!result.isEmpty()) {
+          LOG.warn("Number of failed deletions:{}", result.size());
+          for(Map.Entry<Path, String> singleResult: result) {
+            LOG.info("{}: {}", singleResult.getKey(), singleResult.getValue());
+          }
+        }
       } catch (IllegalArgumentException e) {
-        LOG.error("Caught exception while deleting", e);
-        throw new IOException(e);
+        throw new IOException("Exception while deleting: ", e);
```

Comment:
Assuming e will in turn print path as that was the only pending comment!

Reply:
It will throw from here: 
```
checkArgument(validatePathIsUnderParent(p, basePath),
              "Path %s is not under the base path %s", p, basePath);